packages/aws-cdk/lib/api/util/sdk.ts

```diff
@@ -126,7 +124,19 @@ export class SDK implements ISDK {
       });
     }
 
-    this.endpointCache = new EndpointCache(options.endpointFile);
+    if (options.endpoints) {
+      try {
+        this.endpoints = JSON.parse(options.endpoints);
+      } catch (_e) {
+        this.endpoints = options.endpoints.split(/,/g).reduce((acc, entry) => {
```

Comment:
You can use `.reduce<ServiceEndpoints>((acc, entry) =>` instead of 
`{} as ServiceEndpoints`